rules/plists.bzl

```diff
@@ -1,6 +1,92 @@
+"""
+Defines macros for working with plist files.
+"""
+
 load("@bazel_skylib//lib:types.bzl", "types")
+load("@bazel_skylib//lib:sets.bzl", "sets")
+load("@build_bazel_rules_ios//rules:substitute_build_settings.bzl", "substitute_build_settings")
 load("//rules:library.bzl", "write_file")
-load("//rules/library:xcconfig.bzl", "build_setting_name")
+load("//rules/library:xcconfig.bzl", "build_setting_name", "merge_xcconfigs")
+
+def process_infoplists(name, infoplists, infoplists_by_build_setting, xcconfig, xcconfig_by_build_setting):
+    """
+    Constructs substituted_plists by substituting build settings from an xcconfig dict into the variables of a plist.
+
+    Args:
+        name: The name of the target that the plist is being generated for.
+        infoplists: The plist files to be manipulated
+        infoplists_by_build_setting: A dictionary of infoplists keyed by config_setting, merged with infoplists created here.
+        xcconfig: the default build settings to expand the infoplists with
+        xcconfig_by_build_setting: the build settings grouped by config_setting to expand the infoplists with
+
+    Returns:
+        A selectable dict of the substituted_plists grouped by config_setting
+    """
+    infoplists_by_build_setting = dict(infoplists_by_build_setting)
+    xcconfig_by_build_setting = dict(xcconfig_by_build_setting)
+
+    # Final substituted infoplists_by_build_setting dict
+    substituted_infoplists_by_build_setting = {}
+
+    # Default infoplists are any defined in `infoplists` OR in `infoplists_by_build_setting` with the key `//conditions:default`
+    default_infoplists = infoplists_by_build_setting.pop("//conditions:default", infoplists)
+
+    # Default xcconfig is the one defined in `xcconfig` OR in `xcconfig_by_build_setting` with the key `//conditions:default`
+    default_xcconfig = xcconfig_by_build_setting.pop("//conditions:default", xcconfig)
+
+    # Substitute the default infoplists
+    substituted_infoplists_by_build_setting["//conditions:default"] = [
+        substituted_plist(
+            name = "%s.%s.infoplist_substituted" % (name, idx),
+            plist = plist,
+            xcconfig = default_xcconfig,
+        )
+        for idx, plist in enumerate(write_info_plists_if_needed(name = name, plists = default_infoplists))
+    ]
+
+    # Collect a set of config settings to iterate over
+    config_setting_names = sets.make(infoplists_by_build_setting.keys() + xcconfig_by_build_setting.keys())
+
+    # Substitute the infoplists_by_build_setting
+    for config_setting_name in sets.to_list(config_setting_names):
+        name_suffix = build_setting_name(config_setting_name)
+        infoplists_for_build_setting = infoplists_by_build_setting.get(config_setting_name, default_infoplists)
+        xccconfig_for_build_setting = xcconfig_by_build_setting.get(config_setting_name, {})
+        xcconfig_for_plist = merge_xcconfigs(default_xcconfig, xccconfig_for_build_setting)
```

Comment:
@luispadron The issue I hit with this lines is ambiguous config results.. 

Is there a way to move plist computation / merging till a later point, say analysis phase? That is more robust overall and fixes the issues I have encountered

Imagine you arrive at:
`infoplists_by_build_setting =  { "foo" : [X] }`
`xcconfig_by_build_setting = { "bar" : [Y] }`


But `foo` is actually not mutually exclusive with `bar` from a configuration perspective.. 


I wonder if you can tweak this logic to lift some of the loading phase stuff like `merge_xcconfigs` to the analysis phase.  


For instance:
```
substituted_plist(
   name=name,
   plist=plist,
   base_xcconfig=xcconfig, 
   xcconfig=xcconfig_by_build_setting, 
)
```

Reply:
It also removes merging in the loading phase and making assumptions based on the names 😄